### PR TITLE
Document the yaml_file usage

### PR DESCRIPTION
**Pull request by GaryPEGEOT**
See: https://github.com/symfony/symfony/pull/30446

### CODE REVIEW

File: service_container.rst [hunk 1]
```diff
@@ -1072,6 +1072,44 @@ If you want to pass the second, you'll need to :ref:`manually wire the service <
...
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <!-- ... -->
+
+                <service id="my_service" class="App\ServiceWithYamlInside">
+                    <argument type="yaml_file">%kernel.root_dir%/some/file.yaml</argument>
```
Comment (HeahDude) [inline service_container.rst]: `yaml-file`

File: service_container.rst [hunk 2]
```diff
@@ -1072,6 +1072,44 @@ If you want to pass the second, you'll need to :ref:`manually wire the service <
...
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            # ...
+
+            App\ServiceWithYamlInside:
```
Comment (HeahDude) [inline service_container.rst]: What about `App\ServiceWithArrayConfigFromYamlFile` or something more like it?
The current naming let think that we may get a YAML string.

Comment (GaryPEGEOT) [reply service_container.rst]: Agreed!

File: service_container.rst [hunk 3]
```diff
@@ -1072,6 +1072,44 @@ If you want to pass the second, you'll need to :ref:`manually wire the service <
     and the automatically loaded service will be passed - by default - when you type-hint
     ``SiteUpdateManager``. That's why creating the alias is a good idea.
 
+Injecting the content of YAML file as argument
+----------------------------------------------
+
```
Comment (OskarStark) [inline service_container.rst]: Please add a versionadded directive like for other new features

File: service_container.rst [hunk 4]
```diff
@@ -1072,6 +1072,48 @@ If you want to pass the second, you'll need to :ref:`manually wire the service <
...
+.. versionadded:: 4.3
```
Comment (OskarStark) [inline service_container.rst]: please add a blank line after the directive, thank you

Comment (GaryPEGEOT) [reply service_container.rst]: Done

Comment (OskarStark) [reply service_container.rst]: Thank you

File: service_container.rst [hunk 5]
```diff
@@ -1072,6 +1072,48 @@ If you want to pass the second, you'll need to :ref:`manually wire the service <
...
+                    - !yaml_file '%kernel.root_dir%/some/file.yaml'
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
```
Comment (OskarStark) [inline service_container.rst]: Please use https here

**Review by OskarStark — CHANGES_REQUESTED** [review]

File: service_container.rst (hunk 5 above)
Comment (GaryPEGEOT) [reply service_container.rst]: None of the previous examples use https, sure about that one?

Comment (HeahDude) [reply service_container.rst]: https://github.com/symfony/symfony-docs/pull/11122 is on the go, any addition should be secure now.

Comment (GaryPEGEOT) [reply service_container.rst]: Ok then, fixed!

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Can we merge this?

**Comment by javiereguiluz** [comment]
We can't merge!! The related code PR hasn't been merged, right? https://github.com/symfony/symfony/pull/30446

**Comment by OskarStark** [comment]
Sorry I wanted to comment in the code PR 😊

**Comment by OskarStark** [comment]
The code PR was not merged, so let’s close this one. Anyway thank you for working on this 👍🏻

**Pull request closed without merging**